src/Components/Components/src/RegistrationContext.cs

```diff
@@ -0,0 +1,6 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+internal readonly record struct RegistrationContext(Func<Task> Callback, IComponentRenderMode? RenderMode);
```

Comment:
Can you give this a more specific name? Neither "registration" nor "context" really communicate anything without further context. I know it's only internal but we still have to understand the code :) Suggestion: `PersistComponentStateRegistration`.

Reply:
I've renamed this and changed it for a regular readonly struct, this was leftover from before